code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -112,14 +124,49 @@ public int updateAccount(Account account, String name, String kind, String exter
                 .update(ACCOUNT)
                 .set(properties)
                 .where(ACCOUNT.ID
-                        .eq(account.getId())
-                        .and(ACCOUNT.NAME.eq(account.getName()))
-                        .and(ACCOUNT.EXTERNAL_ID.eq(account.getExternalId())))
-                        .and(ACCOUNT.EXTERNAL_ID_TYPE.eq(account.getExternalIdType()))
-                        .and(ACCOUNT.KIND.eq(account.getKind()))
+                        .eq(account.getId()))
                         .execute();
     }
 
+    @Override
+    public List<AccountRecord> getAccessibleProjects (Long userId, List<String> accessibleIds) {
+        
+        if(null == userId || null == accessibleIds) {
+            return new ArrayList<>();
+        }
+        
+        List<AccountRecord> projects = new ArrayList<>();
+        
+        Result<AccountRecord> orgProjects = create()
```

Comment:
It will take a lot more convincing to make me believe orgs and teams never have overlapping ids.  Besides that this whole function can just be done in one SQL statement.